committee/committee.go

```diff
@@ -163,56 +164,56 @@ func NewCommittee(kvstore db.KVStore, cfg Config) (Committee, error) {
 		interval:              cfg.GravityChainHeightInterval,
 		currentHeight:         0,
 		nextHeight:            cfg.GravityChainStartHeight,
+		syncingChan:           make(chan struct{}, 1),
 	}, nil
 }
-
-func (ec *committee) Start(ctx context.Context) (err error) {
+func (ec *committee) load(ctx context.Context) (err error) {
 	ec.mutex.Lock()
 	defer ec.mutex.Unlock()
 	if err := ec.db.Start(ctx); err != nil {
 		return errors.Wrap(err, "error when starting db")
 	}
-	if startHeight, err := ec.db.Get(db.NextHeightKey); err == nil {
-		zap.L().Info("restoring from db")
-		ec.nextHeight = util.BytesToUint64(startHeight)
-		for height := ec.startHeight; height < ec.nextHeight; height += ec.interval {
-			zap.L().Info("loading", zap.Uint64("height", height))
-			data, err := ec.db.Get(ec.dbKey(height))
-			if err != nil {
-				return err
-			}
-			r := &types.ElectionResult{}
-			if err := r.Deserialize(data); err != nil {
-				return err
-			}
-			ec.cache.insert(height, r)
-			if err := ec.heightManager.add(height, r.MintTime()); err != nil {
-				return err
-			}
-		}
-	}
-	zap.L().Info("catching up via network")
-	tipHeight, err := ec.carrier.TipHeight()
+	startHeight, err := ec.db.Get(db.NextHeightKey)
 	if err != nil {
-		return errors.Wrap(err, "failed to get tip height")
+		return
 	}
-	results, errs := ec.fetchInBatch(tipHeight)
-	if err := ec.storeInBatch(results, errs); err != nil {
-		return errors.Wrap(err, "failed to catch up via network")
+	zap.L().Info("restoring from db")
+	ec.nextHeight = util.BytesToUint64(startHeight)
+	for height := ec.startHeight; height < ec.nextHeight; height += ec.interval {
+		zap.L().Info("loading", zap.Uint64("height", height))
+		data, err := ec.db.Get(ec.dbKey(height))
+		if err != nil {
+			return err
+		}
+		r := &types.ElectionResult{}
+		if err := r.Deserialize(data); err != nil {
+			return err
+		}
+		ec.cache.insert(height, r)
+		if err := ec.heightManager.add(height, r.MintTime()); err != nil {
+			return err
+		}
 	}
+	return nil
+}
+func (ec *committee) Start(ctx context.Context) (err error) {
+	ec.load(ctx)
 	zap.L().Info("subscribing to new block")
 	heightChan := make(chan uint64)
 	reportChan := make(chan error)
+
 	go func() {
 		for {
 			select {
 			case <-ec.terminate:
 				ec.terminate <- true
 				return
 			case height := <-heightChan:
+				ec.syncingChan <- struct{}{}
```

Comment:
This may not be necessary.

heightChan is a single element chan, so that it should prevent multiple sync calls happen in parallel.

Reply:
right，but heightChan will be taken every time ,every time "height := <-heightChan" will be exectute when new height is coming,that means "sync" will be called everytime